orc8r/lib/go/service/logcodec.go

```diff
@@ -24,15 +27,62 @@ import (
 	grpc_proto "google.golang.org/grpc/encoding/proto"
 )
 
+// logModes
+type grpcLogVerbosityLevel int
+
+const (
+	GRPCLOG_DISABLED    = grpcLogVerbosityLevel(iota) // Default value
+	GRPCLOG_FULL                                      // Prints all protps
+	GRPCLOG_HIDEVERBOSE                               // Prints all except verboseProtos
+	grpclog_end
+)
+
+var (
+	// Default list of verbose protos
+	verboseProtos = map[string]grpcLogVerbosityLevel{
+		"protos.Void":                  GRPCLOG_HIDEVERBOSE,
+		"protos.ServiceInfo":           GRPCLOG_HIDEVERBOSE,
+		"protos.HealthStatus":          GRPCLOG_HIDEVERBOSE,
+		"protos.MetricsContainer":      GRPCLOG_HIDEVERBOSE,
+		"protos.SubmitMetricsRequest":  GRPCLOG_HIDEVERBOSE,
+		"protos.SubmitMetricsResponse": GRPCLOG_HIDEVERBOSE,
+		"grpc.MetricFamilies":          GRPCLOG_HIDEVERBOSE,
+		"grpc.Void":                    GRPCLOG_HIDEVERBOSE,
+	}
+)
+
 // logCodec is a debugging Codec implementation for protobuf.
 // It'll be used if debug GRPC printout is enabled
 type logCodec struct {
-	protoCodec encoding.Codec
+	protoCodec     encoding.Codec
+	verbosityLevel grpcLogVerbosityLevel
+}
+
+// Marshal of GRPC Codec interface
+func (lc logCodec) Marshal(v interface{}) ([]byte, error) {
+	printMessage("Sending: ", v, lc.verbosityLevel)
+	return lc.protoCodec.Marshal(v)
+}
+
+// Unmarshal of GRPC Codec interface
+func (lc logCodec) Unmarshal(data []byte, v interface{}) error {
+	err := lc.protoCodec.Unmarshal(data, v)
+	printMessage("Received: ", v, lc.verbosityLevel)
+	return err
+}
+
+// Name of GRPC Codec interface
+func (logCodec) Name() string {
+	return grpc_proto.Name
```

Comment:
those functions have just been moved here. No change except for `printMessage` arguments